app/src/main/java/org/simple/clinic/drugs/selection/custom/drugfrequency/SelectDrugFrequencyDialog.kt

```diff
@@ -0,0 +1,91 @@
+package org.simple.clinic.drugs.selection.custom.drugfrequency
+
+import android.app.Dialog
+import android.content.Context
+import android.content.DialogInterface
+import android.os.Bundle
+import androidx.fragment.app.DialogFragment
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import kotlinx.parcelize.IgnoredOnParcel
+import kotlinx.parcelize.Parcelize
+import org.simple.clinic.R
+import org.simple.clinic.di.injector
+import org.simple.clinic.drugs.search.DrugFrequency
+import org.simple.clinic.drugs.search.DrugFrequency.BD
+import org.simple.clinic.drugs.search.DrugFrequency.OD
+import org.simple.clinic.drugs.search.DrugFrequency.QDS
+import org.simple.clinic.drugs.search.DrugFrequency.TDS
+import org.simple.clinic.drugs.search.DrugFrequency.Unknown
+import org.simple.clinic.navigation.v2.Router
+import org.simple.clinic.navigation.v2.ScreenKey
+import org.simple.clinic.navigation.v2.Succeeded
+import org.simple.clinic.util.unsafeLazy
+import javax.inject.Inject
+
+class SelectDrugFrequencyDialog : DialogFragment() {
+  @Inject
+  lateinit var router: Router
+
+  private val screenKey: Key by unsafeLazy { ScreenKey.key(this) }
+
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+
+    context.injector<Injector>().inject(this)
+  }
+
+  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+    var selectedValueIndex = frequenciesArrayIndexFromDrugFrequency(screenKey.drugFrequency)
+    return MaterialAlertDialogBuilder(requireContext(), R.style.ThemeOverlay_Simple_Dialog)
+        .setTitle(getString(R.string.custom_drug_entry_sheet_frequency))
+        .setSingleChoiceItems(resources.getStringArray(R.array.custom_drug_entry_sheet_frequencies), selectedValueIndex) { _, indexSelected -> selectedValueIndex = indexSelected }
+        .setPositiveButton(getString(R.string.custom_drug_entry_sheet_frequency_dialog_done)) { _, _ ->
+          router.popWithResult(Succeeded(drugFrequencyFromFrequenciesArrayIndex(selectedValueIndex)))
```

Comment:
We can create a list of drug frequencies and use that for rendering the list items and getting frequency based on index.